plugins/hls-hlint-plugin/src/Ide/Plugin/Hlint.hs

```diff
@@ -202,8 +204,11 @@ rules recorder plugin = do
     liftIO $ argsSettings flags
 
   action $ do
-    files <- getFilesOfInterestUntracked
-    void $ uses GetHlintDiagnostics $ Map.keys files
+    files <- Map.keys <$> getFilesOfInterestUntracked
+    Shake.ShakeExtras{ideTesting = Options.IdeTesting testing, lspEnv} <- Shake.getShakeExtras
+    Shake.kickSignal testing lspEnv files (Proxy @"kick/start/hlint")
+    void $ uses GetHlintDiagnostics files
+    Shake.kickSignal testing lspEnv files (Proxy @"kick/done/hlint")
```

Comment:
Couldn't we wrap this to avoid some of this overhead? E.g. `runWithSignal`

```haskell
    runWithSignal "hlint" files $ do
      void $ uses GetHlintDiagnostics files
```

Reply:
sure, uses "kick/start/hlint" ... for now